src/core_plugins/metric_vis/public/metric_vis_controller.js

```diff
@@ -24,12 +30,72 @@ module.controller('KbnMetricVisController', function ($scope, $element, Private)
 
         metrics.push({
           label: column.title,
-          value: value
+          value: value,
+          aggConfig: table.aggConfig(column)
         });
       });
     });
   };
 
+  $scope.clickHandler = function (metric) {
+    let aggConfigResult = null;
+    if (metric.aggConfig.getField() && metric.aggConfig.getField().filterable) {
+      const fieldAggConfig = {
+        schema: {
+          group: 'buckets'
+        },
+        fieldFormatter: metric.aggConfig.fieldFormatter,
+        createFilter: (key) => {
+          return {
+            exists: {
+              field: metric.aggConfig.getField().name
+            },
+            meta: {
+              index: $scope.vis.indexPattern.id
+            }
+          };
+        }
+      };
+      aggConfigResult = new AggConfigResult(fieldAggConfig, aggConfigResult, metric.aggConfig.getField(), 'exists');
+    }
+    if ($scope.vis.getFilters().length > 0) {
```

Comment:
what exactly do you try to achieve with this ? calling `clickHandler` will not remove old filters .... i don't really understand the purpose of this but i think its obsolete.